mscore/scoreview.cpp

```diff
@@ -134,7 +134,16 @@ ScoreView::ScoreView(QWidget* parent)
       double mag  = preferences.getDouble(PREF_SCORE_MAGNIFICATION) * (mscore->physicalDotsPerInch() / DPI);
       _matrix     = QTransform(mag, 0.0, 0.0, mag, 0.0, 0.0);
       imatrix     = _matrix.inverted();
-      _magIdx     = preferences.getDouble(PREF_SCORE_MAGNIFICATION) == 1.0 ? MagIdx::MAG_100 : MagIdx::MAG_FREE;
+      switch (preferences.getInt(PREF_SCORE_ZOOM_TYPE)) {
+            case 0: _magIdx = preferences.getDouble(PREF_SCORE_MAGNIFICATION) == 1.0 ? MagIdx::MAG_100 : MagIdx::MAG_FREE;
+                    break;
+            case 2: _magIdx = MagIdx::MAG_PAGE;
+                    break;
+            case 3: _magIdx = MagIdx::MAG_DBL_PAGE;
+                    break;
+            case 1: // Page Width - also set as default within prefsdialog.ui
+            default: _magIdx = MagIdx::MAG_PAGE_WIDTH;
```

Comment:
Maybe better add a comment here, `// FALLTROUGH` to avoid compiler warnings, or, maybe even better, a `Q_FALLTHROUGH();`  although in this case it might not be strictly needed

Reply:
Yeah, logically I figured default was impossible to get here, but a rule of thumb is always to have a default, so I figured I'd fall through to make it seem like it wasn't redundant.  Didn't explicitly mention it in the comments though...

Actually, now that you mention Q_FALLTHROUGH(), it looks like C++17 has its own [[fallthrough]] which signifies directly to the compiler not to worry about it. I think C++17 is enabled, so let me try that. I'd hate to find out that it fails the mtest though or something...

Reply:
No, we're requiring C++11 only so far

Reply:
Ok. Reverted to C++11 and instead of doing any Q_FALLBACK, just went with the more bare-basic non-fall through with a redundant "just in case" default going to page width. Shouldn't have a problem this way. Thanks.